ngraph/core/src/op/loop.cpp

```diff
@@ -167,16 +168,19 @@ void op::v5::Loop::validate_and_infer_types()
             m_num_iterations = val[0];
     }
 
+    // WA: first input description with index 0 or 1 means that Loop consructor will duplicate it in
+    // the inputs.
+    // When using visit_attributes() no duplication occurs, input_offset shall be decremented.
+    size_t input_offset = 2;
+    if (m_input_descriptions[0].get()->m_input_index == 0 ||
```

Comment:
Looks incorrect. We should check all m_input_descriptions, not only 0.

Not sure, maybe something like that:
```
size_t input_offset = 2;
for (const auto& in_desc :  m_input_descriptions) {
    if (in_desc->m_input_index == 0 || in_desc->m_input_index == 1) {
        input_offset--;
    } 
}

// maybe it's correct to set it to 0, and not throw exception
if (input_offset < 0) {
    // mistake
}
```

Reply:
Probably I got it wrong, in WA description I added that only "first" input_description shall be examined (now it is removed). I agree that we shall check all input descriptions. As for negative `input_offset` I think we shall set it to 0. Is it possible to have 2 or more input descriptions with the same `external_port_id`?